app/views/boxes/_form.haml

```diff
@@ -35,6 +37,28 @@
               name: "box[sample_uuids]", placeholder: "Enter sample id", className: "input-block",
               samples: @box_form.samples_data }
 
+          %fieldset#add_csv{ :disabled => @box_form.option != "add_csv", :style => (@box_form.option == "add_csv" ? "" : "display: none") }
+            #uploaded-files
```

Comment:
The other two cases (`add_batches` and `add_samples`) rely on a React component and the HTML & the whole jQuery-like JS really feels wrong.

Would you be courageous enough to try and refactor to such a basic React component? The two other components are quite basic and merely generate a FORM that will eventually be submitted.

Alternatively, if all we do is select a CSV file and upload it... maybe we could just have the user submit the form, we validate on the backend and render the form again on failure (with the expected error messages). That would avoid an endpoint to validate the CSV, all the custom JS, and feel much more like Rails.

Reply:
Sure, I really like your alternative idea! Shifting the CSV validation to the backend aligns nicely with my preferred style and it simplifies things. It's great that we can avoid having a specialized controller like `csv_validations_controller`. Let's go for it!